mmdet/core/evaluation/eval_hooks.py

```diff
@@ -53,6 +53,8 @@ def _do_evaluate(self, runner):
             return
 
         from mmdet.apis import single_gpu_test
+        # assert when dataloader is empty, in case that Error Occurs when running train.py with the "image" field of json file (val2017.json) set empty
+        assert len(self.dataloader) != 0, "empty dataloader, which may be due to the \"image\" field of json file"
```

Comment:
Adding the assertion directly after building dataloader instead of here would be better.

Reply:
ohhh Thank you. I have posted a new PR refering to your advice #7129